### PR TITLE
Greeting refactoring

### PR DESCRIPTION
**Pull request by viniciusoyama**
Hi, I'm willing to contribute to the project and started playing with the code today. I've decided to start with something simple and created specs / refactored the Greeting service: 

- [x] Using underscore 
- [X] Isolated logic to deal with question quotes
- [X] Trying to improve sayHello method
- [X] Created specs (Following the test style of lib/colors.unit.spec.js file for random cases)
- [X] Fixed a "Potential bug" that when there was no personalization passed to sayHello the method wasn't adding a question mark on the end.

### CODE REVIEW

File: frontend/src/metabase/home/containers/HomepageApp.jsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { connect } from "react-redux";
 import { t } from 'c-3po'
 import { push } from "react-router-redux";
 
-import Greeting from "metabase/lib/greeting";
+import { Greeting } from "metabase/lib/greeting";
```
Comment (kdoh) [inline frontend/src/metabase/home/containers/HomepageApp.jsx]: If we're not going to have Greeting be the default export I'd say we should probably just not bother with an object and have the various methods just be plain functions and do `import { sayHello } from 'metabase/lib/greeting' `

Comment (viniciusoyama) [reply frontend/src/metabase/home/containers/HomepageApp.jsx]: Good idea. Or as we also want to export greetingPrefixes and other things we rename it to Greeter.

Comment (kdoh) [reply frontend/src/metabase/home/containers/HomepageApp.jsx]: Yeah I think plain functions is probably the way to go here.

**Review by kdoh — CHANGES_REQUESTED** [review]
Should convert `Greeting` to just be plain functions. Once that's done and applied in the various places they're used this seems good to me. Thanks for adding tests!

**Review by viniciusoyama — COMMENTED** [review]
Thanks for the quick feedback!

I've changed the implementation to export all functions instead of one object.

One thing I must say is that in order to make it possible for me to test sayHello implementation I added another argument as a dependency to be the greetingGeneratorFn. 

Is that OK? Maybe all of this is just too much for a small feature like this but what I'm trying to do here is to better understand what are the conventions for specs and code here so I can help with more complex features. 

Thanks :)

**Review by salsakran — CHANGES_REQUESTED** [review]
@viniciusoyama Thank you for wanting to help and for writing tests!

In general, we'd prefer that if you want to help out, you start with either writing tests or fixing a bug. Code refactoring for the sake of refactoring isn't really something we're going to merge too quickly. Code churn introduces the possibility of bugs, and takes us time to review. 

The distinction between a greeting object and a set of bare functions has no effect on end users, or really any difference in code maintainability. 

The tests themselves are really too low level and is testing things that seem highly unlikely to break, eg testing that a one line function that calls `_.sample(anArray)` actually returns something in the array isn't especially useful. Also, a having a mock function for such a simple bit of functionality is a bit of a code smell. 

IMO, this can be a simple use of underscore in the same structure + some very simple tests that verify that a greeting is generated. 

We'd much rather you spent your time and ours on things that help our users. If you're interested, I'll look through our issues and find something self-contained that would be a good started bug if you'd like.

**Comment by viniciusoyama** [comment]
Sure! I totally agree with you! 

I'd really appreciate if you help me find some bugs/issues on the frontend that I can work on!

**Comment by kdoh** [comment]
Going to go ahead and close this since it's unlikely to get merged.

**Pull request closed without merging**